exampleProjects/touch/src/main.cpp

```diff
@@ -0,0 +1,127 @@
+//======================== setup pins =======================
+constexpr int TFT_CS { 14 };
+constexpr int TFT_DC { 27 };
+constexpr int pico_SDA { 25 };
+constexpr int pico_SCL { 26 };
+
+//========================= setup screen =======================
+#include <Adafruit_GFX.h>     // Core graphics library
+#include <Adafruit_ILI9341.h> // Hardware-specific library for ST7789
+#include <Adafruit_FT6206.h>
+#include <SPI.h>
+
+Adafruit_ILI9341 tft { TFT_CS, TFT_DC };
+Adafruit_FT6206  ctp;
+
+constexpr int RECT_COUNT    = 16;
+int           SCREEN_HEIGHT = -1;
+int           SCREEN_WIDTH  = -1;
+int           RECT_HEIGHT   = -1;
+
+//======================== setup puara-gestures =======================
+#include <puara/gestures.h>
+puara_gestures::Touch touch;
+int                   discrete_touch[RECT_COUNT] = { 0 };
+
+//======================== main code =======================
+void setup (void)
+{
+    Serial.begin (9600);
+    Serial.println (F ("Touch Puara Gesture Test!"));
+
+    Wire.setPins (pico_SDA, pico_SCL);
+    Wire.begin();
+
+    tft.begin();
+    SCREEN_HEIGHT = tft.height();
+    Serial.println (F ("Screen is Initialized"));
+    SCREEN_WIDTH = tft.width();
+    RECT_HEIGHT  = SCREEN_HEIGHT / RECT_COUNT;
+
+    Serial.println (F ("Touchscreen is initialized"));
+
+    // pass in 'sensitivity' coefficient
+    if (! ctp.begin (40))
+    {
+        Serial.println ("Couldn't start touchscreen controller");
+        while (1)
+            ;
+    }
+    Serial.println ("touchscreen started");
+
+    // Display introductory text
+    tft.fillScreen (ILI9341_BLACK);
+    tft.setTextColor (ILI9341_WHITE);
+    tft.setTextSize (2);
+    tft.setCursor (10, 100);
+    tft.println ("Puara Gesture Test!");
+    tft.setTextSize (1);
+    tft.setCursor (10, 130);
+    tft.println ("Touch the screen to begin.");
+
+    // Wait for a touch before proceeding
+    while (! ctp.touched())
+        delay (50);
+
+    // Draw rectangles
+    tft.fillScreen (ILI9341_BLACK);
+    for (int i = 0; i < RECT_COUNT; i++)
+        tft.drawRect (0, i * RECT_HEIGHT, SCREEN_WIDTH, RECT_HEIGHT, ILI9341_WHITE);
+}
+
+void updateAndPrintTouch()
+{
+    static float prev_touchAll = -1.0f; // Initialize with an invalid value
```

Comment:
I wonder if in terms of app design we woudln't be better with these local static instead be part of a global `Application` class where they would be local member variables, what would you think?

Reply:
might have overdone it lol, lmk what you think!